enhancements/network/egress-ip-multi-nic.md

```diff
@@ -0,0 +1,278 @@
+
+### Risks and Mitigations
+- OVN kube node must recalculate all the matched pods for a given egress IP. This will require refactoring this logic
+  from ovnkube-controller and may introduce bugs to egress IP itself.
+- Stale EgressIPs assigned to a non-ovn managed interface
+  A component will be created to manage all network interface address operations.
+  This component will independently ensure the state of all network interfaces is what we expect.
+  Egress IP(s) assigned to an interface will contain a well-known label which will allow us to clean up stale egress IP addresses.
+  Therefore, we don't need any internal caches for this.
+- Stale linux routing rules in the routing policy database control
+  A component will be created to manage all rule operations.
+  We can ensure stale routes are cleaned because we use a specific priority that only we manage, therefore we don't
+  need any internal caches for this.
+  This component will independently manage the operations of rules and ensure they match the state we want.
+- Stale iptables rules
+  A component will be created to manage all iptable operations.
+  This component will independently manage the operations of iptables and ensure they match the state we want.
+  This component will ensure the rules we expect within a newly created specific chain are what we expect.
+  Therefore, we don't need any internal caches for this because we know we manage all rules within this chain.
+- Incorrect egress IP network interface assignment
+  The host addresses label will determine which network (interface) the egress IP will be assigned to.
+  Users may not expect that we will assign this egress IP to whichever interface contains the longest prefix match [1].
+  This is not a problem because we will emit the packet to the correct network regardless.
+- Unable to determine if an egress IP belongs to an ovn managed network or non ovn manged network
+  OVN managed network can be determined from well-known label `k8s.ovn.org/node-primary-ifaddr`
+  Therefore anything outside this subnet mask is consider a "non-ovn" network
+- What about secondary bridges managed by OVN aka br-ex1
+  We ignore them and do not support egress IP.
+- Selecting the wrong network types from interfaces
+  Only consider addresses with scope `Global`.
+
+[1] https://en.wikipedia.org/wiki/Longest_prefix_match
+
+## Design Details
+### Cluster Manager
+If egress IP is within a subnet of a non-ovn managed network, it will need to assign this egress IP to a node which hosts this network.
+
+### OVN kube controller (aka ovn kube master, aka network controller manager)
+Enhance OVN logical router `ovn_cluster_router` to make the next-hop choice a little smarter. If the egress IP is assigned
+to a secondary network, then reroute it the management port of the node that hosts the secondary network.
+
+### OVN kube node
```

Comment:
What will happen on ovnkube-node when multiple secondary egressIPs assigned to it match the same pod?

Reply:
Undefined right? Today, what happens is undefined.

Reply:
I think it is fair to state at the enhancement level that it is undefined, we can see if this can be addressed in the  implementation

Reply:
Done